src/Methods/BuilderHelper.php

```diff
@@ -163,7 +163,8 @@ public function getMethodReflectionFromBuilder(
         // This can be a custom EloquentBuilder or the normal one
         $builderName = $this->determineBuilderType($modelName);
 
-        $builderReflection = $this->broker->getClass($builderName);
+        /** @var \PHPStan\Reflection\ClassReflection $builderReflection */
+        $builderReflection = (new GenericObjectType($builderName, [new ObjectType($modelName)]))->getClassReflection();
```

Comment:
This change was necessary so that the GenericType is already resolved when calling a builder method statically on a model class.

E.g.
```php
User::create([])
```
Before, its type would be inferred as `Builder<TModelClass of Model>` rather than `Builder<User>`